src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py

```diff
@@ -612,6 +612,142 @@ def test_skipped(self, filename: str, name: str):
         logging.info(f"Skipping test from {filename}: {name}")
 
 
+class PIXITType(Enum):
+    INT = "int"
+    BOOL = "bool"
+    FLOAT = "float"
+    STRING = "string"
+    JSON = "json"
+    HEX = "hex"
+
+    @property
+    def arg_flag(self) -> str:
+        """Maps PIXIT type to command line flag"""
+        return f"--{self.value}-arg"
+
+
+@dataclass
+class PIXITDefinition:
+    """Describes a PIXIT requirement for a test"""
+    name: str
+    pixit_type: PIXITType
+    description: str
+    required: bool = True
+    default: Any = None
+
+    @staticmethod
+    def is_pixit(arg_name: str) -> bool:
+        """Only validate args starting with PIXIT."""
+        return arg_name.startswith("PIXIT.")
+
+
+class PIXITValidator:
+    """Handles validation of PIXIT values against their definitions"""
+
+    @staticmethod
+    def validate_int(value: Any) -> bool:
+        int(value)
+        return True
+
+    @staticmethod
+    def validate_bool(value: Any) -> bool:
+        if isinstance(value, str):
+            value_lower = value.lower()
+            if value_lower not in ('true', 'false', '1', '0', 'yes', 'no'):
+                return False
+        else:
+            bool(value)
+        return True
+
+    @staticmethod
+    def validate_float(value: Any) -> bool:
+        float(value)
+        return True
+
+    @staticmethod
+    def validate_string(value: Any) -> bool:
+        str(value)
```

Comment:
For all these you need to wrap the possibly-failing conversion in a `try: except ValueError` before true/false returned.

Also, when returning False, you are losing the reason for why it failed validation, which would have come out fo the exception.

Recommend moving all of these validators to raise exceptions rather than be bool return. In that case, it could remain `validate_xxx`

Reply:
fixed and converted bool returns to exceptions